models/DataObject/Listing.php

```diff
@@ -140,6 +140,8 @@ public function resetConditionParams()
      */
     public function setCondition($condition, $conditionVariables = null)
     {
+        $this->setData(null);
```

Comment:
This needs to be done for other methods that affect the result as well, such as `addObjectbrick()` `setLocale()` `setIgnoreLocalizedFields()` ....